src/renderer/widgets/WalletDetails/ui/VaultWalletDetails.tsx

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from 'react';
+
+import { BaseModal, ContextMenu, IconButton, HelpText } from '@shared/ui';
+import { useModalClose } from '@shared/lib/hooks';
+import { RootAccountLg, WalletCardLg, VaultAccountsList } from '@entities/wallet';
+import { chainsService } from '@entities/network';
+import { useI18n } from '@app/providers';
+import type { Wallet, BaseAccount } from '@shared/core';
+import { copyToClipboard } from '@shared/lib/utils';
+import { VaultMap } from '../lib/types';
+import { ShardsList } from './ShardsList';
+import { vaultDetailsModel } from '../model/vault-details-model';
+
+type Props = {
+  wallet: Wallet;
+  root: BaseAccount;
+  accountsMap: VaultMap;
+  onClose: () => void;
+};
+export const VaultWalletDetails = ({ wallet, root, accountsMap, onClose }: Props) => {
+  const { t } = useI18n();
+
+  const [isModalOpen, closeModal] = useModalClose(true, onClose);
+
+  const chains = useMemo(() => {
+    return chainsService.getChainsData({ sort: true });
+  }, []);
+
+  return (
+    <BaseModal
+      closeButton
+      contentClass=""
+      panelClass="h-modal"
+      title={t('walletDetails.common.title')}
+      isOpen={isModalOpen}
+      onClose={closeModal}
+    >
+      <div className="flex flex-col w-full">
+        <div className="py-5 px-5 border-b border-divider">
+          <WalletCardLg wallet={wallet} />
+        </div>
+
+        <div className="px-5 py-4">
+          <ContextMenu button={<RootAccountLg name={wallet.name} accountId={root.accountId} />}>
+            <ContextMenu.Group title={t('walletDetails.vault.publicKey')}>
+              <div className="flex items-center gap-x-2">
+                <HelpText className="text-text-secondary break-all">{root.accountId}</HelpText>
+                <IconButton
+                  className="shrink-0"
+                  name="copy"
+                  size={20}
+                  onClick={() => copyToClipboard(root.accountId)}
+                />
+              </div>
+            </ContextMenu.Group>
+          </ContextMenu>
+        </div>
+
+        <VaultAccountsList
+          className="h-[381px]"
+          chains={chains}
+          accountsMap={accountsMap}
+          onShardClick={vaultDetailsModel.events.shardsSelected}
```

Comment:
As I understand I can open only one sharded account at a time. Is it correct?

Reply:
Yes, sure